pkg/daemon/update.go

```diff
@@ -1797,11 +1795,75 @@ func (dn *Daemon) updateSSHKeys(newUsers []ign3types.PasswdUser) error {
 	return nil
 }
 
+// ExtractAndUpdateOS attempts to consolidating various OS update paths to use the same entrypoint here,
+// to move towards handling CoreOS Layering.
+// Does both extraction (old model) and direct updates (new model)
+// Returns the extracted location to remove
+func ExtractAndUpdateOS(newOSImageURL string) (osImageContentDir string, err error) {
+	if isLayeredUpdate(newOSImageURL) {
+		// support new flow here
+		// Not sure if these commands are enough?
```

Comment:
Yep that should be it!

That said...there is a whole interesting discussion here because what we could actually do is rebase *once* to `docker-registry.default.svc:5000/machine-config-operator/coreos-layered:latest` and then just run `rpm-ostree upgrade` to pull the new image.  

OTOH, most of everything inside the platform is all using explicit `@sha256` digests for total predictability.  So doing a `rebase` each time here is fine.